### PR TITLE
Merged from Chris Feature Branch AWS s3 Setup

### PR DESCRIPTION
**Pull request by rswdch**
## Issue Type
- [ ] Bug
- [X] Feature
- [ ] Tech Debt

## Description
* Added driver code to upload a file to s3 bucket `src/services/aws-s3-service.ts` 
* Added node-cron code to schedule backup type (daily/weekly/monthly). 

## Ticket Item
N/A

## Steps to Reproduce Bug / Validate Feature / Confirm Tech Debt Fix
1. Create `cluster-state.yaml` by running `kubectl get all --all-namespaces -o yaml > cluster-config.yaml`
2. Run `npx ts-node src/services/aws-s3-service.ts` 
3. File is uploaded to s3 bucket (need to configure beforehand)

## Previous Behavior
No code for uploading to s3

## Expected Behavior
YAML file of the cluster configuration will be uploaded to s3. However, file is not yet indexed in a database for future recall. 

## Screenshots & Videos
N/A

## Additional Context
* Configure s3 upload in `src/services/aws-s3-service.ts` with appropriate secrets in .env
* TODO UI needs to hook into driver code to be able to save from the UI
* TODO functionality needs to be added to index a reference to the uploaded file in a database for future recall

### CODE REVIEW

File: .env.test
Comment (rswdch) [inline .env.test:1]: Please add to .gitignore

File: cluster-state.yaml
Comment (rswdch) [inline cluster-state.yaml:1]: This might also be good to add to .gitignore since cluster-state.yaml contains secrets

**Comment by rswdch** [comment]
I added some things to .gitignore just in case

**Review by rswdch — COMMENTED** [review]
Made some of my own commits to add to gitignore

**Review by basselfares — APPROVED** [review]

**Comment by rswdch** [comment]
BTW I can't approve because I made the PR for Chris. However, I am OK with approving this.

**Review by nchib — APPROVED** [review]
lgtm

**Pull request merged**